File: assignments/07.08.2023/344.cpp
#include <iostream>

class Solution {
public:
    void reverseString(char* s, int size) {
        int left = 0;
        int right = size - 1;

        while (left < right) {
            std::swap(s[left], s[right]);
            left++;
            right--;
        }
    }
};

int main() {
    char s[] = "hello";
    int size = sizeof(s) - 1; // Exclude the null terminator

    Solution solution;
    std::cout << "Original string: " << s << std::endl;
    solution.reverseString(s, size);
    std::cout << "Reversed string: " << s << std::endl;

    return 0;
}

File: assignments/07.08.2023/344.java
public class Solution {
    public void reverseString(char[] s) {
        int left = 0;
        int right = s.length - 1;
        
        while (left < right) {
            char temp = s[left];
            s[left] = s[right];
            s[right] = temp;
            left++;
            right--;
        }
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        char[] input = {'h', 'e', 'l', 'l', 'o'};
        System.out.println("Original: " + new String(input));
        
        solution.reverseString(input);
        System.out.println("Reversed: " + new String(input));
    }
}
